File: docs/guides/swift/Sources/enableFilterPromote.swift
import Foundation
#if os(Linux) // For linux interop
    import FoundationNetworking
#endif

import Core
import Search

func enableFilterPromote() async throws {
    let condition = SearchCondition(
        pattern: "{facet:brand}",
        anchoring: .is
    )

    let consequence = SearchConsequence(
        filterPromotes: true
    )

    let rule = Rule(objectID: "rule_with_filterPromotes", conditions: [condition], consequence: consequence)
    print(rule)
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/enableFilterPromote.kt
import com.algolia.client.configuration.*
import com.algolia.client.extensions.*
import com.algolia.client.model.search.*
import com.algolia.client.transport.*

suspend fun enableFilterPromote() {
  val condition = Condition(pattern = "{facet:brand}", anchoring = Anchoring.Is)

  val consequence = Consequence(filterPromotes = true)

  val rule =
    Rule(
      objectID = "rule_with_filterPromotes",
      conditions = listOf(condition),
      consequence = consequence,
    )
}
